src/mpeg4/mpeg4BoxHeader.ts

```diff
@@ -0,0 +1,262 @@
+     * Gets the total size of the box described by the current instance.
+     * @returns A value containing the total size of the box described by the current instance.
+     */
+    public get totalBoxSize(): number {
+        return this._boxSize;
+    }
+
+    /**
+     * Gets the position box represented by the current instance in the file it comes from.
+     * @returns A value containing the position box represented by the current instance in the file it comes from.
+     */
+    public get position(): number {
+        return this._fromDisk ? this._position : -1;
+    }
+
+    /**
+     * Constructs and initializes a new instance of @see Mpeg4BoxHeader that is empty.
+     * @returns A new instance of @see Mpeg4BoxHeader that is empty.
+     */
+    public static fromEmpty(): Mpeg4BoxHeader {
+        return Mpeg4BoxHeader.fromType(ByteVector.fromString("xxxx", StringType.UTF8));
+    }
+
+    /**
+     * Constructs and initializes a new instance of @see Mpeg4BoxHeader by reading it from a specified seek position in a specified file.
+     * @param file A @see File object to read the new instance from.
+     * @param position A value specifying the seek position in File at which to start reading.
+     * @returns A new instance of @see Mpeg4BoxHeader by reading it from a specified seek position in a specified file.
+     */
+    public static fromFileAndPosition(file: File, position: number): Mpeg4BoxHeader {
+        Guards.notNullOrUndefined(file, "file");
+
+        const header = new Mpeg4BoxHeader();
+        header.box = undefined;
+        header._fromDisk = true;
+        header._position = position;
+        file.seek(position);
+
+        const data: ByteVector = file.readBlock(32);
+        let offset: number = 0;
+
+        if (data.length < 8 + offset) {
+            throw new Error("Not enough data in box header.");
+        }
+
+        header._headerSize = 8;
+        header._boxSize = data.subarray(offset, 4).toUint();
+        header._boxType = data.subarray(offset + 4, 4);
+
+        // If the size is 1, that just tells us we have a massive ULONG size waiting for us in the next 8 bytes.
+        if (header._boxSize === 1) {
+            if (data.length < 8 + offset) {
+                throw new Error("Not enough data in box header.");
+            }
+
+            header._headerSize += 8;
+            offset += 8;
+            header._boxSize = Number(data.subarray(offset, 8).toUlong());
+        }
+
+        // UUID has a special header with 16 extra bytes.
+        if (ByteVector.equals(header.boxType, Mpeg4BoxType.Uuid)) {
+            if (data.length < 16 + offset) {
+                throw new Error("Not enough data in box header.");
+            }
+
+            header._headerSize += 16;
+            header._extendedType = data.subarray(offset, 16);
+        } else {
+            header._extendedType = undefined;
+
+            if (header._boxSize > file.length - position) {
+                throw new Error(
```

Comment:
I agree we wouldn't want a guard for these. I'd consider making this a `CorruptFileError` since the data in the file says the box is x bytes long, and since there's not enough bytes in the file to read the box, it'd make the file unusable to us. But, there's kind of a mix of error types all over the place in the code, so this is likely splitting hairs. As long as an error gets returned here, it doesn't matter too much what type it is.

Reply:
Ok understood. So I assume I keep the error as it is?

Reply:
Yep, that's fine by me